url3f8d6440dd_huiwq1990_CognativeModel_tgz-pJ8-clarion_samples_HelloWorldJ8/src/clarion/samples/XOR.java

```diff
@@ -0,0 +1,343 @@
+		 * sensory input space.*/
+		CombinedInputSpace = new DimensionValueCollection (SensoryInputSpace);
+		//END initialization of the sensory input space.
+	}
+
+	/**
+	 * This method reports the results of each agent during the simulation.
+	 */
+	public void reportResults() {
+		Accuracy = CorrectCounter/NumberTrials;
+		CorrectCounter = 0;
+		double percentage = Accuracy*100.0;
+		BigDecimal bd = new BigDecimal( Double.toString(percentage) );
+		bd = bd.setScale( DecimalPlace, BigDecimal.ROUND_HALF_UP );
+
+		if (CurrentTest == NumberTestRuns){
+			System.out.println(" |"+bd.doubleValue()+"|- run complete");
+			if (useVerboseLogging){
+				System.out.println("The agent learned the following rules through RER:");
+				for(AbstractRule r : Agent.getACS().getRERRules())
+				{
+					//Report the rules the agent learned using RER.
+					System.out.println(r.toString());		
+				}
+			}
+		}
+		else
+			System.out.print(" |"+bd.doubleValue()+"| ");
+			
+	}
+		
+	/**
+	 * This method reports the final results after all agents have been run.
+	 */
+	public void reportFinalResults(){
+		System.out.println("All tests and trials complete.");
+		double percentAccuracy = (AvgAccuracyNumerator/NumberAgents)*100.0;
+		CorrectCounter = 0;
+		BigDecimal bd = new BigDecimal( Double.toString(percentAccuracy) );
+		bd = bd.setScale( DecimalPlace, BigDecimal.ROUND_HALF_UP );
+		
+		System.out.println("Number of Trials per Test: "+ NumberTrials);
+		System.out.println("Number of Tests per Run: "+NumberTestRuns);
+		System.out.println("Number of Trials per Run:"+ NumberTestRuns*NumberTrials);
+		System.out.println("Number of Agents: "+ NumberAgents);
+		System.out.println("Avg (accuracy after all tests) over all agents: "+ bd.doubleValue());
+	}
+	
+	/**
+	 * This method runs an agent through the XOR task.
+	 */
+	public void run(CLARION Agent) {
+		//For keeping track of the feedback.
+		int feedback = 0;
+		//For keeping track of the chosen action.
+		AbstractAction ChosenAction;
+		//Run the task for the specified number of trials.
+		
+		for(CurrentTest = 1;CurrentTest<=NumberTestRuns;CurrentTest++){
+			for(int i = 0; i < NumberTrials; i++)
+			{
+				//Randomly choose an input to give.
+				if(Math.random() > .5)
+				{
+					if(Math.random()>.5)
+					{
+					//Specify the input to give as "True True".
+						CombinedInputSpace.get("Boolean1").get("True").setActivation(1);
+						CombinedInputSpace.get("Boolean1").get("False").setActivation(0);
+						CombinedInputSpace.get("Boolean2").get("True").setActivation(1);
+						CombinedInputSpace.get("Boolean2").get("False").setActivation(0);
+				
+					}else{
+						//Specify the input to give as "False False".
+						CombinedInputSpace.get("Boolean1").get("True").setActivation(0);
+						CombinedInputSpace.get("Boolean1").get("False").setActivation(1);
+						CombinedInputSpace.get("Boolean2").get("True").setActivation(0);
+						CombinedInputSpace.get("Boolean2").get("False").setActivation(1);
+				
+					}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 253:268, 271:288.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.